soroban-sdk/src/crypto/bls12_381.rs

```diff
@@ -84,6 +88,14 @@ pub struct Fp(BytesN<48>);
 #[repr(transparent)]
 pub struct Fp2(BytesN<96>);
 
+/// `Fr` represents an element in the BLS12-381 scalar field, which is a prime
+/// field of order `r` (the order of the G1 and G2 groups). The struct is
+/// internally represented with an `U256`, all arithmetic operations follow
+/// modulo `r`.
+#[derive(Clone)]
+#[repr(transparent)]
+pub struct Fr(U256);
+
 impl_bytesn_repr!(G1Affine, 96);
 impl_bytesn_repr!(G2Affine, 192);
 impl_bytesn_repr!(Fp, 48);
```

Comment:
Can we add a conversion to BytesN? Or does that not make sense?

Reply:
yeah it makes sense. added conversion to/from BytesN is via U256.

Reply:
@jayz22 Could you add a `impl_bytesn_rep!(Fr, 32)`, instead of implementing the two functions on their own?

Reply:
The `impl_bytesn_rep` is intended for a `BytesN` wrapping type, here the `Fr` is wrapping an `U256` and the only extra methods needed are conversion from bytes?

Reply:
Or you want me to add conversions to/from array (`[u8;32]`) as well, this are the only extra methods `impl_bytesn_rep` provides (the val-conversion methods are already implemented via U256).

Reply:
Oh, sorry, yes you're right.

Reply:
This is fine, lets merge. Feel free to add the follow up to/from, but I'll merge it now to avoid churning because this is good.

Reply:
I assume the bytes are accessible via U256 anyway. Sorry for the above, I forgot it wasn't wrapping a BytesN.